packages/trader/src/App/Components/Elements/market-countdown-timer.jsx

```diff
@@ -0,0 +1,203 @@
+            const diff = +getUTCDate(open[i]) - +new Date();
+            if (diff > 0) {
+                remaining_time_to_open = +getUTCDate(open[i]);
+                return { days_offset, opening_time: open[i], remaining_time_to_open };
+            }
+        }
+        return whenMarketOpens(days_offset + 1, target_symbol);
+    }
+};
+
+const calculateTimeLeft = remaining_time_to_open => {
+    const difference = remaining_time_to_open - +new Date();
+    let timeLeft = {};
+
+    if (difference > 0) {
+        timeLeft = {
+            days: Math.floor(difference / (1000 * 60 * 60 * 24)),
+            hours: Math.floor((difference / (1000 * 60 * 60)) % 24),
+            minutes: Math.floor((difference / 1000 / 60) % 60),
+            seconds: Math.floor((difference / 1000) % 60),
+        };
+    }
+
+    return timeLeft;
+};
+
+const padWithZero = number => {
+    if (number < 10) {
+        return `0${number}`;
+    }
+    return number;
+};
+
+const MarketCountdownTimer = ({ is_main_page, setActiveSymbols, symbol }) => {
+    const [when_market_opens, setWhenMarketOpens] = React.useState({});
+    const [time_left, setTimeLeft] = React.useState(calculateTimeLeft(when_market_opens?.remaining_time_to_open));
+
+    React.useEffect(() => {
+        let is_subscribed = true;
+        async function fetchTradingTimes() {
+            const result = await whenMarketOpens(0, symbol);
+            if (is_subscribed) setWhenMarketOpens(result);
+        }
+        fetchTradingTimes();
+
+        return () => (is_subscribed = false);
+    }, [symbol]);
+
+    React.useEffect(() => {
+        let timer;
+        if (when_market_opens?.remaining_time_to_open) {
+            timer = setTimeout(async () => {
+                setTimeLeft(calculateTimeLeft(when_market_opens?.remaining_time_to_open));
+                if (when_market_opens.remaining_time_to_open <= 1 && typeof setActiveSymbols === 'function')
+                    await setActiveSymbols();
+            }, 1000);
+        }
+        return () => {
+            if (timer) {
+                clearTimeout(timer);
+            }
+        };
+    }, [time_left, when_market_opens]);
+
+    let timer_components = '';
+
+    Object.keys(time_left).forEach(interval => {
+        if (interval === 'days') {
+            if (time_left.days) {
+                timer_components += `${time_left.days} ${localize(time_left.days > 1 ? 'days' : 'day')} `;
+            }
+        } else {
+            const value = time_left[interval];
+            timer_components += interval !== 'seconds' ? `${padWithZero(value)}:` : `${padWithZero(value)}`;
+        }
+    });
+
+    const { opening_time, days_offset } = when_market_opens;
+    let opening_time_banner = null;
+    if (is_main_page && opening_time) {
+        const date_target = new Date();
+        date_target.setDate(date_target.getDate() + days_offset);
+        const date = moment(date_target).format('D MMM YYYY:dddd');
```

Comment:
let stick the `moment` library, you can use `moment(). add(days_offset,'days')` this way I think we only use one thing that makes it more readbale